### PR TITLE
use backend object as default backend instead of backend in proxy

### PR DESCRIPTION
**Pull request by mkudlej**

### CODE REVIEW

**Comment by mkudlej** [comment]
Still there are failing sandbag tests:
ERROR testsuite/tests/apicast/parameters/test_apicast_service.py::test_filter_by_url
ERROR testsuite/tests/apicast/parameters/test_apicast_services_list.py::test_apicast_services_list_param
ERROR testsuite/tests/apicast/parameters/apicast_path_routing/test_apicast_path_routing.py::test_not_mapped_route_returns_not_found
ERROR testsuite/tests/apicast/parameters/apicast_path_routing/test_apicast_path_routing_query.py::test_non_matching_args
ERROR testsuite/tests/apicast/parameters/apicast_path_routing/test_mapping_rule_wrongly_matched.py::test_mapping_rule_wrongly_matched
--> draft

File: testsuite/tests/conftest.py [hunk 1]
```diff
@@ -862,7 +875,8 @@ def finalizer():
             if backends:
                 for path, backend in backends.items():
                     svc.backend_usages.create({"path": path, "backend_api_id": backend["id"]})
-                proxy_params = {}
+                proxy_params.pop("api_backend", None)
+                proxy_params.pop("api_test_path", None)
```
Comment (mganisin) [inline testsuite/tests/conftest.py:879]: An addition of these `pop()` calls is breaking change

**Review by mganisin — APPROVED** [review]

**Comment by mkudlej** [comment]
All tests passed -> from draft

File: testsuite/tests/conftest.py [hunk 2]
```diff
@@ -811,7 +824,7 @@ def __init__(self):
         def _custom_service(
             self,
             params,
-            proxy_params=None,
+            proxy_params={},
```
Comment (pehala) [inline testsuite/tests/conftest.py:827]: This is wrong, you should not have a mutable default value.

Comment (mkudlej) [reply testsuite/tests/conftest.py]: You're faster than pylint!!!

**Review by dhlavac — DISMISSED** [review]

**Review by JaurbanRH — APPROVED** [review]

**Comment by mkudlej** [comment]
After removing all implicit backends all tests passed except of 
```
testsuite/tests/prometheus/system/test_internal_calls.py
testsuite/tests/prometheus/backend_listener/test_backend_listener_internal_api.py
testsuite/tests/apicast/parameters/jaeger/test_open_telemetry_apicast_integration.py
``` but these failures are not related to this change.

File: testsuite/tests/apicast/apiap/test_proxy_config.py [hunk 3]
```diff
@@ -15,11 +15,11 @@
...
 
 @pytest.fixture(scope="module")
-def service(custom_service, service_settings):
+def service(custom_service, service_settings, private_base_url):
     """
-    Create a custom service withou backend
+    Create a custom service without backend
     """
-    return custom_service(service_settings)
+    return custom_service(service_settings, proxy_params={"api_backend": private_base_url()}, backends={})
```
Comment (JaurbanRH) [inline testsuite/tests/apicast/apiap/test_proxy_config.py:22]: No need to add proxy params and empty backends

File: testsuite/tests/apicast/parameters/http_proxy/large_data/test_http_proxy_large_data.py [hunk 4]
```diff
@@ -19,10 +19,13 @@
 
 
 @pytest.fixture(scope="module")
-def service_proxy_settings(private_base_url, protocol):
-    """Url for service that is configured with http version of echo api"""
+def backend_default(private_base_url, custom_backend, protocol):
+    """
+    Default backend with url from private_base_url.
+    Url for service that is configured with http version of echo api.
```
Comment (JaurbanRH) [inline testsuite/tests/apicast/parameters/http_proxy/large_data/test_http_proxy_large_data.py:25]: `echo api` in docstring and httpbin_go in test ?

File: testsuite/tests/apicast/apiap/test_proxy_config.py [hunk 5]
```diff
@@ -17,9 +17,9 @@
 @pytest.fixture(scope="module")
 def service(custom_service, service_settings):
     """
-    Create a custom service withou backend
+    Create a custom service without backend
     """
-    return custom_service(service_settings)
+    return custom_service(service_settings, proxy_params={}, backends={})
```
Comment (JaurbanRH) [inline testsuite/tests/apicast/apiap/test_proxy_config.py:22]: Please remove `proxy_params` and `backends` arguments here.

File: testsuite/mockserver.py [hunk 6]
```diff
@@ -82,5 +82,5 @@ def verify_sequence(self, expected_requests) -> bool:
             verify=self.verify,
         )
         if response.status_code == 400:
-            raise HTTPError("Invalid matcher format", response=response)
+            raise HTTPError("Invalid matcher format", response=response)  # type: ignore
```
Comment (JaurbanRH) [inline testsuite/mockserver.py:85]: Unrelated change

**Comment by mkudlej** [comment]
It is ready for merge, but I'm going to move it into draft because we want to start to use it at beginning of new cycle.

**Review by JaurbanRH — APPROVED** [review]

**Review by mdujava — CHANGES_REQUESTED** [review]
Rebase for openshift client upgrade and added pytest "<8.0.0"

**Review by mdujava — APPROVED** [review]
